src/services/tts.py

```
@@ -0,0 +1,124 @@
+    mp3_bytes = _pcm_to_mp3(pcm_bytes, sample_rate)
+    if mp3_bytes is None:
+        return None, None
+
+    duration = _probe_duration(mp3_bytes)
+    return mp3_bytes, duration
+
+
+def _pcm_to_mp3(pcm_bytes: bytes, sample_rate: int) -> bytes | None:
+    """Transcode raw PCM s16le to MP3 via ffmpeg."""
+    try:
+        result = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)